docs/proposals/control-plane-config.md

```diff
@@ -17,3 +17,94 @@ option that we will support is log level.
 ## Non-Goals
 
 This proposal is *not* defining a way to dynamically configure the data plane.
+
+## Introduction
+
+The NKG control plane will evolve to have various user-configurable options. These could include, but are not
+limited to, log level, tracing, or metrics. For the best user experience, these options should be able to be
+changed at runtime, to avoid having to restart NKG. The first option that we will allow users to configure is the
+log level. The easiest and most intuitive way to implement a Kubernetes-native API is through a CRD.
+
+In this doc, the term "user" will refer to the cluster operator (the person who installs and manages NKG). The
+cluster operator owns this CRD resource.
+
+## API, Customer Driven Interfaces, and User Experience
+
+The API would be provided in a CRD. An authorized user would interact with this CRD using `kubectl` to `get`
```

Comment:
Forgot to add a comment. Would it make sense to explain why we don't put in as reference in the GatewayClass resource? So that we capture this "why" in this doc,  and also why we chose to reference this resource from a cli arg among the alternatives?